libflux/flux-core/src/ast/tests.rs

```diff
@@ -2131,3 +2131,48 @@ fn test_ast_json_roundtrip() {
     };
     assert_eq!(ast, roundtrip_ast);
 }
+
+/// Locator makes constructing an SourceLocation from a string simple.
+pub struct Locator<'a> {
```

Comment:
I don't know how you feel about a dependency here, but influxdata/flux-lsp uses [line_col](https://docs.rs/line-col/0.2.1/line_col/) for this and, as I recall, it handles some cases that aren't the happy path, e.g. different line endings, etc.

_gets farther down in the patch_ Ah, I see, it's just moved code. Still, wanted to keep this comment to see what your thoughts were.